### PR TITLE
[SPARK-37805][TESTS] Refactor `TestUtils#configTestLog4j` method to use log4j2 api

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
SPARK-37795 add a scalastyle rule to ban `org.apache.log4j` imports, but there is still one place to retain the imports of `org.apache.log4j` in `o.a.spark.TestUtils`.

This pr refactor `configTestLog4j` method in `o.a.spark.TestUtils` to use log4j2 api and let the log behavior using log4j2.x be the same as that using log4j1.x before. In fact, the `configTestLog4j` method behavior before this pr is invalid because `PropertyConfigurator.configure` method in `org.apache.logging.log4j:log4j-1.2-api` is an empty method as follows:

https://github.com/apache/logging-log4j2/blob/491a0b3787975b6fc95b6a8cb3da76dc7517c65f/log4j-1.2-api/src/main/java/org/apache/log4j/PropertyConfigurator.java#L39-L47

Another change of this pr is rename the method name from `configTestLog4j`  to `configTestLog4j2`.

### Why are the changes needed?
Clean up the `org.apache.log4j` imports left in Spark internal and let `configTestLog4j` method behavior keep consistent between log4j1.x and log4j2.x.  

### Does this PR introduce _any_ user-facing change?
The  `configTestLog4j` method in `TestUtils` rename to `configTestLog4j2`

### How was this patch tested?

- Pass the Jenkins or GitHub Action
- Manual test

Run the test cases using `configTestLog4j` method in the following 3 scenarios:

1. without this pr to test log4j2.x
2. with this pr to test log4j2.x
3. run `git reset --hard 19227983e91a54b5f27ade6412dad1088dfcba9e` to test log4j1.x

For example `WholeStageCodegenSparkSubmitSuite`, run 

```
mvn clean install -DskipTests -pl sql/core -am
mvn test -pl sql/core -Dtest=none -DwildcardSuites=org.apache.spark.sql.execution.WholeStageCodegenSparkSubmitSuite
```

Scenario 1 does not print any logs to the console, scenario 2 and scenario 3 will print similar logs to the console

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
cc @viirya

**Comment by dongjoon-hyun** [comment]
FYI, @LuciferYang . It was intentionally ignored at that time after discussing with @williamhyun offline.

**Comment by LuciferYang** [comment]
> FYI, @LuciferYang . It was intentionally ignored at that time after discussing with @williamhyun offline.

What is the specific reason? I didn't find it in SPARK-37795

**Comment by LuciferYang** [comment]
GA run failed，Let me check it first

**Comment by LuciferYang** [comment]
GA Passed

File: core/src/main/scala/org/apache/spark/TestUtils.scala [hunk 1]
```diff
@@ -418,17 +419,18 @@ private[spark] object TestUtils {
...
+  def configTestLog4j2(level: String): Unit = {
+    val builder = ConfigurationBuilderFactory.newConfigurationBuilder()
+    val appenderBuilder = builder.newAppender("console", "CONSOLE")
+      .addAttribute("target", ConsoleAppender.Target.SYSTEM_ERR)
+    appenderBuilder.add(builder.newLayout("PatternLayout")
+      .addAttribute("pattern", "%d{yy/MM/dd HH:mm:ss} %p %c{1}: %m%n"))
+    builder.add(appenderBuilder)
+    builder.add(builder.newRootLogger(s"$level").add(builder.newAppenderRef("console")))
+    val configuration = builder.build()
+    LogManager.getContext(false).asInstanceOf[LoggerContext].reconfigure(configuration)
```
Comment (viirya) [inline core/src/main/scala/org/apache/spark/TestUtils.scala:433]: This rewriting looks okay. I just wonder the purpose of `configTestLog4j` and if it is necessary. These tests are not checking the log (as they are still passed with no-op `configTestLog4j` now). And seems it is not used to reduce verbose logging.

Comment (viirya) [reply core/src/main/scala/org/apache/spark/TestUtils.scala:433]: Maybe we can just remove `configTestLog4j`?

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/TestUtils.scala:433]: I traced back to the original pr ([SPARK-3193](https://github.com/apache/spark/pull/2108)) it seems that `configTestLog4j ` is only added to help troubleshoot the failed UT on Jenkins.

I agree to remove this method directly, what do you think @dongjoon-hyun ?

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/TestUtils.scala:433]: If we are sure to remove the `configTestLog4j `directly, I will update the pr and pr description

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/TestUtils.scala:433]: Hmm... It seems that the method of calling `configTestLog4j` in the current UTs is in the process of `runSparkSubmit`.  If we directly remove this method, I can not find out where to see the relevant log of `runSparkSubmit` (not in `target/units`), which is really not conducive to troubleshooting.

Comment (viirya) [reply core/src/main/scala/org/apache/spark/TestUtils.scala:433]: Then it is okay to keep it, I think.

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/TestUtils.scala:433]: ok ~

File: core/src/main/scala/org/apache/spark/TestUtils.scala [hunk 2]
```diff
@@ -418,17 +419,18 @@ private[spark] object TestUtils {
...
-      "%d{yy/MM/dd HH:mm:ss} %p %c{1}: %m%n")
-    PropertyConfigurator.configure(pro)
+  def configTestLog4j2(level: String): Unit = {
+    val builder = ConfigurationBuilderFactory.newConfigurationBuilder()
+    val appenderBuilder = builder.newAppender("console", "CONSOLE")
+      .addAttribute("target", ConsoleAppender.Target.SYSTEM_ERR)
+    appenderBuilder.add(builder.newLayout("PatternLayout")
+      .addAttribute("pattern", "%d{yy/MM/dd HH:mm:ss} %p %c{1}: %m%n"))
+    builder.add(appenderBuilder)
+    builder.add(builder.newRootLogger(s"$level").add(builder.newAppenderRef("console")))
```
Comment (viirya) [inline core/src/main/scala/org/apache/spark/TestUtils.scala:431]: `s"$level"` -> `level`. `level` is already a string, no need to interpolate it.

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/TestUtils.scala]: cc06076 fix this

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/TestUtils.scala]: @viirya Any other need change?

Comment (viirya) [reply core/src/main/scala/org/apache/spark/TestUtils.scala]: I'm not sure about https://github.com/apache/spark/pull/35095#issuecomment-1004685780.

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/TestUtils.scala]: thank @viirya , ping @dongjoon-hyun @williamhyun

**Review by srowen — CHANGES_REQUESTED** [review]
Looks OK pending tests

**Comment by LuciferYang** [comment]
8d0a078 merge with master

**Comment by LuciferYang** [comment]
> Looks OK pending tests

GA passed

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by LuciferYang** [comment]
thanks all

**Pull request closed without merging**